File: src/ros2_bridge/mappers/visualization_msgs/interactive_marker_pose.rs
//! Typed mapper for `visualization_msgs/msg/InteractiveMarkerPose`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn interactive_marker_pose_to_bus(
    msg: ros_env::visualization_msgs::msg::InteractiveMarkerPose,
) -> crate::visualization_msgs::msg::v1::InteractiveMarkerPose {
    crate::visualization_msgs::msg::v1::InteractiveMarkerPose {
        header: Some(crate::ros2_bridge::mappers::std_msgs::header::header_to_bus(msg.header)),
        pose: Some(crate::ros2_bridge::mappers::geometry_msgs::pose::pose_to_bus(msg.pose)),
        name: crate::ros2_bridge::mappers::convert::from_ros_string(msg.name),
    }
}

pub(crate) fn interactive_marker_pose_to_ros(
    bus: crate::visualization_msgs::msg::v1::InteractiveMarkerPose,
) -> ros_env::visualization_msgs::msg::InteractiveMarkerPose {
    ros_env::visualization_msgs::msg::InteractiveMarkerPose {
        header: crate::ros2_bridge::mappers::std_msgs::header::header_to_ros(
            bus.header.unwrap_or_default(),
        ),
        pose: crate::ros2_bridge::mappers::geometry_msgs::pose::pose_to_ros(
            bus.pose.unwrap_or_default(),
        ),
        name: crate::ros2_bridge::mappers::convert::to_ros_string(bus.name),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct VisualizationMsgsInteractiveMarkerPoseMapper;

impl TypedTopicMapper for VisualizationMsgsInteractiveMarkerPoseMapper {
    type Ros = ros_env::visualization_msgs::msg::InteractiveMarkerPose;
    type Bus = crate::visualization_msgs::msg::v1::InteractiveMarkerPose;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(interactive_marker_pose_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(interactive_marker_pose_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/visualization_msgs/interactive_marker_pose.py
"""Generated mapper for `visualization_msgs/msg/InteractiveMarkerPose`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.std_msgs.header import header_to_bus, header_to_ros
from robot_bus.ros2_bridge.mappers.geometry_msgs.pose import pose_to_bus, pose_to_ros

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.visualization_msgs.msg.v1 import InteractiveMarkerPose as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def interactive_marker_pose_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.header.CopyFrom(header_to_bus(msg.header))
    bus.pose.CopyFrom(pose_to_bus(msg.pose))
    bus.name = str(msg.name)
    return bus


def interactive_marker_pose_to_ros(bus):
    from visualization_msgs.msg import InteractiveMarkerPose as RosMsg

    out = RosMsg()
    out.header = header_to_ros(bus.header)
    out.pose = pose_to_ros(bus.pose)
    out.name = str(bus.name)
    return out


class VisualizationMsgsInteractiveMarkerPoseMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from visualization_msgs.msg import InteractiveMarkerPose as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return interactive_marker_pose_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return interactive_marker_pose_to_ros(bus)
